src/llmq/context.cpp

```diff
@@ -20,108 +20,81 @@
+        return llmq::quorumManager.get();
+    }()},
+    sigman{std::make_unique<llmq::CSigningManager>(connman, *llmq::quorumManager, peerman, unit_tests, wipe)},
+    shareman{std::make_unique<llmq::CSigSharesManager>(connman, *llmq::quorumManager, *sigman, peerman)},
+    clhandler{[&]() -> llmq::CChainLocksHandler* const {
+        llmq::chainLocksHandler = std::make_unique<llmq::CChainLocksHandler>(mempool, connman, sporkman, *sigman, *shareman, *llmq::quorumManager, *::masternodeSync, peerman);
+        return llmq::chainLocksHandler.get();
+    }()},
+    isman{[&]() -> llmq::CInstantSendManager* const {
+        llmq::quorumInstantSendManager = std::make_unique<llmq::CInstantSendManager>(mempool, connman, sporkman, *llmq::quorumManager, *sigman, *shareman, *llmq::chainLocksHandler, *::masternodeSync, peerman, unit_tests, wipe);
+        return llmq::quorumInstantSendManager.get();
+    }()}
+{
     // NOTE: we use this only to wipe the old db, do NOT use it for anything else
     // TODO: remove it in some future version
     auto llmqDbTmp = std::make_unique<CDBWrapper>(unit_tests ? "" : (GetDataDir() / "llmq"), 1 << 20, unit_tests, true);
 }
 
-void LLMQContext::Destroy() {
+LLMQContext::~LLMQContext() {
+    // LLMQContext doesn't own these objects, but still need to care of them for consistancy:
     llmq::quorumInstantSendManager.reset();
     llmq::chainLocksHandler.reset();
-    shareman.reset();
-    sigman.reset();
     llmq::quorumManager.reset();
-    qdkgsman.reset();
     llmq::quorumBlockProcessor.reset();
-    dkg_debugman.reset();
-    bls_worker.reset();
     {
         LOCK(llmq::cs_llmq_vbc);
         llmq::llmq_versionbitscache.Clear();
     }
 }
 
 void LLMQContext::Interrupt() {
-    if (shareman != nullptr) {
-        shareman->InterruptWorkerThread();
-    }
-    if (llmq::quorumInstantSendManager != nullptr) {
-        llmq::quorumInstantSendManager->InterruptWorkerThread();
-    }
+    shareman->InterruptWorkerThread();
+
+    assert(isman == llmq::quorumInstantSendManager.get());
+    llmq::quorumInstantSendManager->InterruptWorkerThread();
 }
 
 void LLMQContext::Start() {
-    if (bls_worker != nullptr) {
-        bls_worker->Start();
-    }
-    if (qdkgsman != nullptr) {
-        qdkgsman->StartThreads();
-    }
-    if (llmq::quorumManager != nullptr) {
-        llmq::quorumManager->Start();
-    }
-    if (shareman != nullptr) {
-        shareman->RegisterAsRecoveredSigsListener();
-        shareman->StartWorkerThread();
-    }
-    if (llmq::chainLocksHandler != nullptr) {
-        llmq::chainLocksHandler->Start();
-    }
-    if (llmq::quorumInstantSendManager != nullptr) {
-        llmq::quorumInstantSendManager->Start();
-    }
+    assert(quorum_block_processor == llmq::quorumBlockProcessor.get());
+    assert(qman == llmq::quorumManager.get());
+    assert(clhandler == llmq::chainLocksHandler.get());
+    assert(isman == llmq::quorumInstantSendManager.get());
```

Comment:
these could still be null which could cause crashes below.. Are we sure these are always non-null? It seems yes as they are assigned in the constructor.

Reply:
yes, if `this` exists  than means all members are correctly initialized and not nullptr.